src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -73,36 +73,40 @@ public class IndexStore : IDisposable
                                                  IndexCompiler.IndexExtension.Length -
                                                  4; // ".dll"
 
-        public readonly DatabaseIndexLockModeController LockMode;
+        private readonly DocumentDatabase _documentDatabase;
 
-        public readonly DatabaseIndexPriorityController Priority;
+        private readonly CollectionOfIndexes _indexes = new CollectionOfIndexes();
+        private readonly ConcurrentDictionary<string, SemaphoreSlim> _indexLocks = new ConcurrentDictionary<string, SemaphoreSlim>(StringComparer.OrdinalIgnoreCase);
 
-        public readonly DatabaseIndexStateController State;
+        private bool _initialized;
 
-        public readonly DatabaseIndexDeleteController Delete;
+        private bool _run = true;
 
-        public readonly DatabaseIndexCreateController Create;
+        private long _lastSurpassedAutoIndexesDatabaseRecordEtag;
 
-        public readonly DatabaseIndexHasChangedController HasChanged;
+        private const int PathLengthLimit = 259; // Roslyn's MetadataWriter.PathLengthLimit = 259
```

Comment:
Can we revert all the place shifting above, it seems to be not necessary.

Reply:
fixed